src/controllers/articles.controller.js

```diff
@@ -73,24 +73,10 @@ class Articles {
    */
   static async getAllArticles(req, res) {
     const counter = await db.count();
-    let page = parseInt(req.query.page, 10);
-    if (isNaN(page) || page < 1) {
-      page = 1;
+    if (req.offset >= counter) {
+      req.offset = 0;
     }
-    let limit = parseInt(req.query.limit, 10);
-    if (isNaN(limit)) {
-      limit = 10;
-    } else if (limit > 50) {
-      limit = 50;
-    } else if (limit < 1) {
-      limit = 1;
-    }
-    let offset = (page - 1) * limit;
-    if (offset >= counter) {
-      offset = 0;
-    }
-
-    const { searchQueries } = req;
+    const { searchQueries, offset, limit } = req;
     const articles = await articleService.getAllArticles(offset, limit, searchQueries);
     if (!articles) {
       return res.status(200).json({ status: 200, message: 'There is no article.' });
```

Comment:
Clarification on status codes that was raised by Leon applies here

Reply:
Should be 404.

Reply:
That's great